src/index.ts

```diff
@@ -67,17 +67,17 @@ export default class ImageTool implements BlockTool {
   /** 
    * Uploader module instance 
    */
-  private uploader: Uploader;
+  private uploader: Uploader<AdditionalUploadResponse>;
 
   /** 
    * UI module instance
    */
-  private ui: Ui;
+  private ui: Ui<CustomActions, AdditionalUploadResponse>;
```

Comment:
I think it would be better to pass `ImageToolData<CustomActions, AdditionalUploadResponse>` because UI should not depend on AdditionalUploadResponse, it depends on data format instead since it may be used for rendering